src/diff.jl

```diff
@@ -37,3 +41,19 @@ differentiate(p::RationalPoly, x::PolyVar) = (differentiate(p.num, x) * p.den -
 function differentiate{C}(p::PolyType{C}, xs::Vector{PolyVar{C}})
     [differentiate(p, x) for x in xs]
 end
+
+# In Julia v0.5, it returns Any
+Base.promote_op{C}(differentiate, ::Type{PolyVar{C}}, ::Type{PolyVar{C}}) = Term{true, Int}
```

Comment:
My concern was that it's not used in the function body anywhere, which seemed strange if it wasn't being used for dispatch. I think it's fine, though; dispatch should work fine based on the other arguments.